deployments/polygon/usdc/migrations/1681313581_more_comp.ts

```diff
@@ -0,0 +1,78 @@
+import { Contract } from 'ethers';
+import { DeploymentManager, migration } from '../../../../plugins/deployment_manager';
+import { calldata, exp, proposal } from '../../../../src/deploy';
+
+import { expect } from 'chai';
+
+const ERC20PredicateAddress = '0x40ec5B33f54e0E8A33A975908C5BA1c14e5BbbDf';
+const RootChainManagerAddress = '0xA0c68C638235ee32657e8f720a23ceC1bFc77C77';
+
+export default migration('1661899622_rewards', {
+  async prepare(deploymentManager: DeploymentManager) {
+    return {};
+  },
+
+  async enact(deploymentManager: DeploymentManager, govDeploymentManager: DeploymentManager) {
+    const trace = deploymentManager.tracer();
+    const ethers = deploymentManager.hre.ethers;
+
+    const {
+      governor,
+      comptrollerV2,
+      COMP,
+    } = await govDeploymentManager.getContracts();
+
+    const {
+      rewards,
+    } = await deploymentManager.getContracts();
+
+    const RootChainManager = await deploymentManager.existing(
+      'RootChainManager',
+      RootChainManagerAddress
+    );
+    const COMPAmountToBridge = exp(10_000, 18); // XXX
+    const depositCOMPData = ethers.utils.defaultAbiCoder.encode(['uint256'], [COMPAmountToBridge]);
+    const depositForCOMPCCalldata = ethers.utils.defaultAbiCoder.encode(
+      ['address', 'address', 'bytes'],
+      [rewards.address, COMP.address, depositCOMPData]
+    );
+
+    const actions = [
+      // 4. Approve Polygon's ERC20Predicate to take Timelock's COMP (for bridging)
+      {
+        contract: COMP,
+        signature: 'approve(address,uint256)',
+        args: [ERC20PredicateAddress, COMPAmountToBridge]
+      },
+      // 5. Bridge COMP from mainnet to Polygon CometRewards using RootChainManager
+      {
+        target: RootChainManager.address,
+        signature: 'depositFor(address,address,bytes)',
+        calldata: depositForCOMPCCalldata
+      },
+    ];
+    const description = "# Refresh COMP Rewards\n"; // XXX
+    const txn = await deploymentManager.retry(
+      async () => governor.propose(...await proposal(actions, description))
+    );
+    trace(txn);
+
+    const event = (await txn.wait()).events.find(event => event.event === 'ProposalCreated');
+    const [proposalId] = event.args;
+    trace(`Created proposal ${proposalId}.`);
+  },
+
+  async verify(deploymentManager: DeploymentManager) {
+    const {
+      rewards,
+    } = await deploymentManager.getContracts();
+
+    // 4. & 5.
+    const polygonCOMP = new Contract(
+      '0x8505b9d2254A7Ae468c0E9dd10Ccea3A837aef5c',
+      ['function balanceOf(address account) external view returns (uint256)'],
+      deploymentManager.hre.ethers.provider
+    );
```

Comment:
You should be able to get COMP directly from `deploymentManager.getContracts()` now that it is part of roots.

Reply:
Yeah I didn't love this but saw we had done this way in the other one, will try

Reply:
Yeah the reason was because rewards were not turned on yet before that proposal so COMP was not spidered. Another way we could've done it was to spider in the `verify` block (that's what we currently do for the Optimism, which I think is cleaner).